src/appleseed/foundation/meta/tests/test_imagefilewriter.cpp

```diff
@@ -0,0 +1,219 @@
+// appleseed.foundation headers.
+#include "foundation/image/color.h"
+#include "foundation/image/colorspace.h"
+#include "foundation/image/genericimagefilereader.h"
+#include "foundation/image/genericimagefilewriter.h"
+#include "foundation/image/image.h"
+#include "foundation/image/imageattributes.h"
+#include "foundation/image/pixel.h"
+#include "foundation/math/scalar.h"
+#include "foundation/utility/iostreamop.h"
+#include "foundation/utility/test.h"
+
+// Standard headers.
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <cstring>
+#include <exception>
+#include <memory>
+
+using namespace foundation;
+
+/*
+// Default export formats:
+//
+// OpenEXR   .exr          4-channel   16-bit (half)                            Linear
+// RGBE      .hdr          3-channel   32-bit (8-bit RGB + shared exponent)     Linear
+// TIFF      .tiff/.tif    4-channel   16-bit (std::uint16_t)                   Linear
+// BMP       .bmp          4-channel    8-bit (std::uint8_t)                    sRGB
+// PNG       .png          4-channel    8-bit (std::uint8_t)                    sRGB
+// JPEG      .jpg/.jpe/    3-channel    8-bit (std::uint8_t)                    sRGB
+//           .jpeg/.jif/
+//           .jfif/.jfi
+*/
+
+TEST_SUITE(Foundation_Image_ImageFileWriter)
+{
+    bool write_image(const std::string ImageFilePath)
+    {
+        const std::string FileExtension = ImageFilePath.substr(ImageFilePath.find_last_of("."));
+    
+        Image image(16, 16, 16, 16, 4, PixelFormatUInt8); 
+        image.clear(Color4b(50, 100, 150, 42));
+        
+        GenericImageFileWriter writer(ImageFilePath.c_str());
+        writer.append_image(&image);
+        writer.write();
+                        
+        return true;
+    }
+    
+    bool read_and_compare_RGBA(const std::string ImageFilePath)
+    {
+        
+        const std::string FileExtension = ImageFilePath.substr(ImageFilePath.find_last_of("."));
+        
+        GenericImageFileReader reader;
+        std::unique_ptr<Image> image(reader.read(ImageFilePath.c_str()));
+        
+        for (size_t y = 0; y < 16; ++y)
```

Comment:
The code that checks every pixel should stay in the test itself and you should use EXPECT_ instead of returning a value. Otherwise, how can you tell which pixel is wrong ? 

Though, you can make a method that simply read the image and return a `std::unique_ptr<Image>`. But that's not really useful.